### PR TITLE
Scale motors

### PR DESCRIPTION
**Pull request by hbradio**
* Refactors motor stuff to a class
* Adds a scaling table to avoid the lower-power motor settings. We were seeing an issue where motor speeds below 35 were getting "missed". Maybe there just wasn't enough power to get the wheels moving?

### CODE REVIEW

File: inc/drivers/GigglebotMotors.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#ifndef GIGGLEBOT_MOTORS_H
+#define GIGGLEBOT_MOTORS_H
+
+#include "MicroBitComponent.h"
+#include "MicroBitI2C.h"
+
+#define GIGGLEBOT_MOTORS_EVT_UPDATE         1
```
Comment (cabarnes) [inline inc/drivers/GigglebotMotors.h]: I don't think this is used

File: inc/drivers/GigglebotMotors.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+class GigglebotMotors
+{
+    MicroBitI2C i2c;
+
+    public:
+    GigglebotMotors(MicroBitI2C &_i2c);
```
Comment (cabarnes) [inline inc/drivers/GigglebotMotors.h]: What if this took the motor number. Then main could construct a left motor, right motor, and both motors object from this and call setMotorPower on the specific one. setMotorPower in main could go away

File: inc/drivers/GigglebotMotors.h (hunk 1 above)
Comment (hbradio) [reply inc/drivers/GigglebotMotors.h]: Good catch.

File: inc/drivers/GigglebotMotors.h (hunk 2 above)
Comment (hbradio) [reply inc/drivers/GigglebotMotors.h]: Good idea. And I'll change the class name to singular.

Comment (hbradio) [reply inc/drivers/GigglebotMotors.h]: ✔️

File: inc/drivers/GigglebotMotors.h (hunk 1 above)
Comment (hbradio) [reply inc/drivers/GigglebotMotors.h]: ✔️

File: inc/drivers/GigglebotMotor.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+class GigglebotMotor
+{
+    MicroBitI2C i2c;
+    const static short int SCALED_POWERS[];
+
+    public:
+    GigglebotMotor(MicroBitI2C &_i2c, GigglebotMotorId _motorId);
+    GigglebotMotorId motorId;
```
Comment (cabarnes) [inline inc/drivers/GigglebotMotor.h]: This doesn't need to be public

**Review by cabarnes — APPROVED** [review]

**Pull request merged**